### PR TITLE
[Dinomod] Add Qianzhousaurus to Dinoclysm

### PR DESCRIPTION
**Pull request by Theawesomeboophis**
Summary
SUMMARY: Mods "Add Qianzhousaurus as well as baby and zombie variants."
Testing
Loads properly, looks right, right color, all variants tested.

### CODE REVIEW

**Comment by Theawesomeboophis** [comment]
Previous Commits tried to add orcs as well, orcs are coming btw, still working on them.

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 1]
```diff
@@ -1252,4 +1252,44 @@
     "placate_triggers": [ "MEAT" ],
     "categories": [ "DINOSAUR" ]
   }
+  {
+    "type": "MONSTER",
+    "id": "mon_Qianzhousaurus",
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monsters/dinosaur.json:1257]:
```suggestion
    "id": "mon_qianzhousaurus",
```
ID formatting is supposed to be all lowercase.

File: data/mods/DinoMod/monsters/hatchling.json [hunk 2]
```diff
@@ -674,5 +674,18 @@
       "SWIMS",
       "AQUATIC"
     ]
+  },
+        {
+    "id": "mon_Qianzhousaurus_hatchling",
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monsters/hatchling.json:679]:
```suggestion
    "id": "mon_qianzhousaurus_hatchling",
```

File: data/mods/DinoMod/monsters/hatchling.json [hunk 3]
```diff
@@ -674,5 +674,18 @@
...
+    "type": "MONSTER",
+    "name": "Baby Qianzhousaurus",
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monsters/hatchling.json:681]:
```suggestion
    "name": "baby Qianzhousaurus",
```

File: data/mods/DinoMod/monstergroups/wilderness.json [hunk 4]
```diff
@@ -1030,6 +1030,20 @@
         "cost_multiplier": 30,
         "pack_size": [ 1, 2 ],
         "conditions": [ "DAY", "SPRING", "SUMMER", "AUTUMN" ]
+        {
+        "monster": "mon_Qianzhousaurus",
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monstergroups/wilderness.json:1034]:
```suggestion
        "monster": "mon_qianzhousaurus",
```

File: data/mods/DinoMod/monstergroups/wilderness.json [hunk 5]
```diff
@@ -1030,6 +1030,20 @@
...
+        "freq": 50,
+        "cost_multiplier": 0,
+        "pack_size": [ 4, 8 ],
+        "conditions": [ "DUSK", "DAWN", "SPRING", "SUMMER", "AUTUMN" ]
+         {
+        "monster": "mon_Qianzhousaurus_juvenile",
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monstergroups/wilderness.json:1040]:
```suggestion
        "monster": "mon_qianzhousaurus_juvenile",
```

File: data/mods/DinoMod/monstergroups/zinosaur.json [hunk 6]
```diff
@@ -173,6 +173,7 @@
       { "monster": "mon_zentaceratops", "freq": 1, "cost_multiplier": 50, "starts": 72, "pack_size": [ 1, 2 ] },
       { "monster": "mon_zorosaurus", "freq": 1, "cost_multiplier": 50, "starts": 72, "pack_size": [ 1, 2 ] },
       { "monster": "mon_zriceratops", "freq": 1, "cost_multiplier": 60, "starts": 72, "pack_size": [ 1, 2 ] }
+      { "monster": "mon_Zianzhousaurus", "freq": 4, "cost_multiplier": 70, "starts": 72, "pack_size": [ 4, 8 ] }
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monstergroups/zinosaur.json:176]:
```suggestion
      { "monster": "mon_zianzhousaurus", "freq": 4, "cost_multiplier": 70, "starts": 72, "pack_size": [ 4, 8 ] }
```

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 7]
```diff
@@ -1252,4 +1252,44 @@
...
+    "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 3 },
+    "special_attacks": [ [ "Poke", 20 ], { "type": "hit", "cooldown": 10 } ],
+    "flags": [ "SEES", "HEARS", "ANIMAL", "PATH_AVOID_DANGER_1", "DESTROYS", "WARM", "CATFOOD", "SWARMS", "ATTACKMON", "HIT_AND_RUN" ],
+    "harvest": "dino_large_pred",
+    "anger_triggers": [ "STALK", "PLAYER_WEAK", "HURT" ],
+    "placate_triggers": [ "MEAT" ],
+    "categories": [ "DINOSAUR" ],
+    "type": "MONSTER",
+  }, 
+
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monsters/dinosaur.json:1294]:
```suggestion
  }
```

File: data/mods/DinoMod/monsters/zed-dinosaur.json [hunk 8]
```diff
@@ -836,4 +836,26 @@
     "description": "A small four-legged plant-eating dinosaur juvenile corpse with natural bone armor plates.",
     "upgrades": { "half_life": 14, "into": "mon_zalamosaurus" }
   }
+   {
+    "type": "MONSTER",
+    "id": "mon_Zianzhousaurus",
+    "name": { "str": "Zombie Qianzhousaurus" },
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monsters/zed-dinosaur.json:842]:
```suggestion
    "id": "mon_Zianzhousaurus",
    "name": { "str": "Qianz zombie" },
```

File: data/mods/DinoMod/monsters/zed-dinosaur.json [hunk 9]
```diff
@@ -836,4 +836,26 @@
...
+    "speed": 175,
+    "melee_skill": 10,
+    "melee_dice": 6,
+    "melee_dice_sides": 7,
+    "melee_cut": 6,
+    "armor_bash": 10,
+    "armor_cut": 10,
+    "armor_bullet": 20,
+    "hp": 230,
+    "description": " An undead Qianzhousaurus, it looks at you with a furious rage in it's eyes and seems intent on eating you.",
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monsters/zed-dinosaur.json:855]:
```suggestion
    "description": "An undead Qianzhousaurus, looking at you with a furious rage in its eyes and seeming intent on eating you.",
```

**Review by LaVeyanFiend — CHANGES_REQUESTED** [review]
ID formatting issues, also throw all the files in [the linter.](https://dev.narc.ro/cataclysm/format.html)

**Comment by Theawesomeboophis** [comment]
Thanks, I'll fix these lines today.

**Comment by Theawesomeboophis** [comment]
Just commited changes, thanks btw.

**Comment by LyleSY** [comment]
Looks like your SUMMARY up top is failing formatting requirements. Try adding the word "Mods" between "SUMMARY:" and "Add"
It should look like this:

#### Summary

SUMMARY: Mods "Add Qianzhousaurus as well as baby and zombie variants."

**Comment by Theawesomeboophis** [comment]
Changed all instances to lowercase.

**Comment by Theawesomeboophis** [comment]
And changed summary to recommended.

**Comment by Theawesomeboophis** [comment]
How do I fix json validation?

**Comment by LyleSY** [comment]
Some of it is linting, just copy your code, paste it here https://dev.narc.ro/cataclysm/format.html
...hit the lint button, then copy the results without the extra blank lines to use instead with nice clean formatting. If it throws an error there's probably a stray comma like I noted above.

**Comment by Theawesomeboophis** [comment]
It's a syntax error, how do I fix that?

**Review by LyleSY — CHANGES_REQUESTED** [review]
I've noted the little easy ones. Big picture, please make sure the size and weight of the dino and zombie variant are as realistic as possible, that the speed is same, and I would look at T Rex and Z Rex as a guide on stats. That's the closest comparison I can think of.

**Comment by LyleSY** [comment]
There's a full list of JSON style issues to look for here https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_STYLE.md

File: data/mods/DinoMod/monstergroups/wilderness.json [hunk 10]
```diff
@@ -1030,6 +1030,20 @@
...
+        "pack_size": [ 4, 8 ],
+        "conditions": [ "DUSK", "DAWN", "SPRING", "SUMMER", "AUTUMN" ]
+         {
+        "monster": "mon_qianzhousaurus_juvenile",
+        "freq": 20,
+        "cost_multiplier": 10,
+        "pack_size": [ 4, 12 ],
+        "conditions": [ "SPRING", "SUMMER", "AUTUMN" ]
+      },
+      },
```
[the remark this answers is not in the document]
Comment (Theawesomeboophis) [reply data/mods/DinoMod/monstergroups/wilderness.json]: No, I'll Make it spawn in wilderness and Lab.

File: data/mods/DinoMod/monstergroups/zinosaur.json [hunk 11]
```diff
@@ -173,6 +173,7 @@
       { "monster": "mon_zentaceratops", "freq": 1, "cost_multiplier": 50, "starts": 72, "pack_size": [ 1, 2 ] },
       { "monster": "mon_zorosaurus", "freq": 1, "cost_multiplier": 50, "starts": 72, "pack_size": [ 1, 2 ] },
       { "monster": "mon_zriceratops", "freq": 1, "cost_multiplier": 60, "starts": 72, "pack_size": [ 1, 2 ] }
+      { "monster": "mon_zianzhousaurus", "freq": 4, "cost_multiplier": 70, "starts": 72, "pack_size": [ 4, 8 ] }
```
[the remark this answers is not in the document]
Comment (Theawesomeboophis) [reply data/mods/DinoMod/monstergroups/zinosaur.json:176]: Didn't Realize It spawned in cities, gonna fix that and make it spawn in forest.

File: data/mods/DinoMod/monsters/juvenile.json [hunk 12]
```diff
@@ -635,6 +635,31 @@
...
+      "NO_BREED",
+      "CANPLAY"
+    ]
+  },
     ]
   }
 ]
+    
+  
+
```
[the remark this answers is not in the document]
Comment (Theawesomeboophis) [reply data/mods/DinoMod/monsters/juvenile.json:665]: How do I delete those lines?

[the remark this answers is not in the document]
Comment (Theawesomeboophis) [reply data/mods/DinoMod/monsters/juvenile.json]: Nevermind.

**Comment by LyleSY** [comment]
It looks like you're not defining a Qianzhousaurus  egg here. That will need to be added to egg.json to spawn dinos without an error

**Comment by Theawesomeboophis** [comment]
Ok, also made it so Zianzhousaurus spawns in forest, not cities.

**Comment by Theawesomeboophis** [comment]
I didn't mean for zianzhousaurus to spawn in cities so I fixed it( I think, it might still be wrong).

**Comment by Theawesomeboophis** [comment]
Added eggs, forgot to add them earlier today.

**Comment by Theawesomeboophis** [comment]
json validation still failing, must have missed something.

**Comment by LyleSY** [comment]
Please review the Design doc https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/DinoMod/DESIGN.md
No big dangerous predators or zombies in the forest, we don't want to murder players on day 1 with something they can't fight. Not a lore thing, just gameplay balance
I'd recommend spawning this in the dinosaur spawn groups, specifically GROUP_DINOSAUR and GROUP_DINOSAUR_DANGEROUS and maybe the hatchling in GROUP_DINOSAUR_DANGEROUS_HATCHLING and the juvenile in GROUP_DINOSAUR_DANGEROUS_JUVENILE . 
Jumping into the lore, the reason dinos are coming back is that portals to parallel Earths are popping up as part of the Cataclyms and dinos are crossing over into our world. Science and Swamper locations can grab dinos from anywhere on Earth because they are/were actively going out and grabbing samples and moving them around and messing with them, but dinos in the wilderness groups should be North American dinos, at least until we change the lore again. The old lore was anything goes because they only spawned in swamps and players don't go there anyway.

**Comment by Theawesomeboophis** [comment]
Where should I make them spawn, in order to balance gameplay?

**Comment by LyleSY** [comment]
Looks like a comma missing before the new egg
image

**Comment by Theawesomeboophis** [comment]
Ok, I'll add the comma.

**Comment by Theawesomeboophis** [comment]
Should they spawn in lab only?

**Comment by Theawesomeboophis** [comment]
Looks like I forgot another comma in the hatchlings.json

**Comment by LyleSY** [comment]
I was thinking about the dinosaur spawn lists here https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/DinoMod/monstergroups/dinosaur.json
...but I would support lab spawns too, sure

**Comment by Theawesomeboophis** [comment]
So I should remove from wilderness.json and move to lab.json, right?

**Comment by Theawesomeboophis** [comment]
My mistake, comma is missing in the egg groups json not hatchling json.

**Comment by Theawesomeboophis** [comment]
Where should Zianzhousaurus spawn, and how do I get it to spawn in that location?

**Comment by LyleSY** [comment]
GROUP_DINOSAUR_ZOMBIE would be a good place for it. That will add it to DinoMod locations and quests

**Comment by Theawesomeboophis** [comment]
I can't track down the validator error.

**Comment by Theawesomeboophis** [comment]
It has seasons and I don't know where it is.

**Comment by LyleSY** [comment]
Looks like two files need linting and wherever you're spawning juveniles needs a comma before the juveniles
image

**Comment by Theawesomeboophis** [comment]
Thanks, I couldn't find it.

**Comment by Theawesomeboophis** [comment]
That's in eggs json right?

**Comment by LyleSY** [comment]
Just checked, it's in wilderness

**Comment by Theawesomeboophis** [comment]
It was a problem in labs too.

**Comment by Theawesomeboophis** [comment]
I seem to have added commas to wilderness.

**Comment by Theawesomeboophis** [comment]
Fixing Qianzhousaurus in wilderness has for some reason broken Eoraptor.

**Comment by Theawesomeboophis** [comment]
Gonna try and copy old unbroken Eoraptor to file.

**Comment by Theawesomeboophis** [comment]
Nope, Eoraptor still broken, must be spacing problem or something similar.

**Comment by LyleSY** [comment]
Oh I see. It's treating the "{" leading in to eoraptor as an unknown character. Deleting that and replacing it in a JSON editor should fix

**Comment by LyleSY** [comment]
I swear this gets easier

File: data/mods/DinoMod/monstergroups/wilderness.json [hunk 13]
```diff
@@ -12,6 +12,19 @@
         "pack_size": [ 4, 8 ],
         "conditions": [ "DUSK", "DAWN", "SPRING", "SUMMER", "AUTUMN" ]
       },
+        {
+        "monster": "mon_qianzhousaurus",
+        "freq": 50,
+        "cost_multiplier": 0,
+        "pack_size": [ 4, 8 ],
+        "conditions": [ "DUSK", "DAWN", "SPRING", "SUMMER", "AUTUMN" ]
+         {
```
Comment (Qrox) [inline data/mods/DinoMod/monstergroups/wilderness.json:21]:
```suggestion
         },
         {
```
Missing a closing bracket here.

File: data/mods/DinoMod/monsters/hatchling.json [hunk 14]
```diff
@@ -674,5 +674,17 @@
...
     ]
+  },
+        {
+    "id": "mon_qianzhousaurus_hatchling",
+    "type": "MONSTER",
+    "name": "baby Qianzhousaurus",
+    "default_faction": "qianzhousaurus",
+   "upgrades": { "age_grow": 40, "into": "mon_qianzhousaurus_juvenile" }
+  },
+    ]
```
Comment (Qrox) [inline data/mods/DinoMod/monsters/hatchling.json:685]:
```suggestion
```

File: data/mods/DinoMod/monsters/juvenile.json [hunk 15]
```diff
@@ -634,7 +634,29 @@
...
+      "PATH_AVOID_DANGER_1",
+      "WARM",
+      "CATFOOD",
+      "GUILT",
+      "STUMBLES",
+      "NO_BREED",
+      "CANPLAY"
+    ]
+  },
     ]
```
Comment (Qrox) [inline data/mods/DinoMod/monsters/juvenile.json:660]:
```suggestion
```

File: data/mods/DinoMod/monsters/zed-dinosaur.json [hunk 16]
```diff
@@ -836,4 +836,25 @@
...
+    "melee_cut": 6,
+    "armor_bash": 10,
+    "armor_cut": 10,
+    "armor_bullet": 20,
+    "hp": 230,
+    "description": "An undead Qianzhousaurus, looking at you with a furious rage in its eyes and seeming intent on eating you.",
+    "upgrades": { "half_life": 14, "into_group": "GROUP_Zianzhousaurus_UPGRADE" },
+    "flags": [ "SEES", "SMELLS", "HEARS", "BASHES", "POISON", "STUMBLES", "NO_BREATHE", "REVIVES", "FILTHY", "WARM" ]
+  },
+
```
Comment (Qrox) [inline data/mods/DinoMod/monsters/zed-dinosaur.json:859]:
```suggestion
  }
```

**Comment by Theawesomeboophis** [comment]
Applied suggested changes.

**Comment by Theawesomeboophis** [comment]
xargs -0 -L 1 tools/format/json_formatter.cgi
Needs linting : data/mods/DinoMod/egg.json
Please read doc/JSON_STYLE.md
Needs linting : data/mods/DinoMod/monstergroups/monstergroups_egg.json
Please read doc/JSON_STYLE.md
terminate called after throwing an instance of 'JsonError'
  what():  Json error: <unknown source file>:76:8: separator not strictly allowed at end of array

        "pack_size": [ 4, 12 ],
        "conditions": [ "SPRING", "SUMMER", "AUTUMN" ]
      },
       ^

    ]
  }

xargs: tools/format/json_formatter.cgi: terminated by signal 6
grep: write error: Broken pipe
find: ‘standard output’: Broken pipe
find: write error
make: *** [style-json] Error 123
Makefile:1149: recipe for target 'style-json' failed
Error: Process completed with exit code 2.

**Comment by Theawesomeboophis** [comment]
json validation should (hopefully) be fixed now.

**Comment by Theawesomeboophis** [comment]
oops, forgot another comma.

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 17]
```diff
@@ -1251,5 +1251,44 @@
...
+    "melee_dice_sides": 8,
+    "melee_cut": 16,
+    "armor_bash": 12,
+    "armor_cut": 12,
+    "armor_bullet": 12,
+    "hp": 45,
+	"vision_day": 36,
+    "death_function": [ "NORMAL" ],
+    "zombify_into": "mon_zianzhousaurus",
+    "description": " Also known as pinnochio rex, this bipedal dinosaur is slower than most and covered in fur, it also looks very hungry.",
```
Comment (Qrox) [inline data/mods/DinoMod/monsters/dinosaur.json:1282]:
```suggestion
    "description": "Also known as pinnochio rex, this bipedal dinosaur is slower than most and covered in fur, it also looks very hungry.",
```

**Review by Qrox — COMMENTED** [review]
Please lint the changes.

**Comment by LyleSY** [comment]
Whew, that is a powerful error
image

Please just revert wilderness back to where it is in the current game. That will disappear that file from these checks. The rest need linting.

**Comment by Theawesomeboophis** [comment]
Linted all files that needed linting.

**Comment by Theawesomeboophis** [comment]
That doesn't look good.

**Comment by Theawesomeboophis** [comment]
How do I restore it to the previous version? I messed up linting.

**Comment by LyleSY** [comment]
Not sure about your specific setup. Personally I just use the website so less can go wrong. I would just copy and lint the files, hit that edit button, drop the linted data in and submit each one like normal. That should clear the conflicts if I understand the situation correctly

**Comment by Qrox** [comment]
I think you used the wrong linting tool. You lint a file by going to `http://dev.narc.ro/cataclysm/format.html`, paste the entire file into the text box, click lint, and copy the result back to replace the content of the file.

**Comment by LyleSY** [comment]
Okay, looks like this needs to be rebased to fix. That's super complicated. Are you on the Discord? https://discord.gg/jFEc7Yp  We can try and walk you through it in dev-help there.

**Comment by Theawesomeboophis** [comment]
I'll try relinting first, and I'm not a discord person

**Comment by Theawesomeboophis** [comment]
Do you have a link to a tutorial for rebasing in case relint doesn't work?

**Comment by Theawesomeboophis** [comment]
I'll work on this over the weekend and try and get it in a mergeable state.

**Comment by Theawesomeboophis** [comment]
Almost erased conflicts by using proper linting website, just have to fix eggs.

**Comment by Theawesomeboophis** [comment]
All conflicts eliminated.

**Comment by Theawesomeboophis** [comment]
Json validation should pass this time.

**Comment by Theawesomeboophis** [comment]
Thanks for the help btw.

**Comment by Theawesomeboophis** [comment]
Hatchlings need relint.

**Comment by Theawesomeboophis** [comment]
Relint seems to have killed conflicts, didn't use the right linter.

**Comment by Theawesomeboophis** [comment]
json validation finally passed.

**Comment by Theawesomeboophis** [comment]
Thx for helping, sorry this took so long, I have no coding experience at all.

**Comment by LyleSY** [comment]
You're great, this is how I learned too. It seriously does get way easier, just the first time sucks

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 18]
```diff
@@ -1251,5 +1251,54 @@
...
+  },
+  {
+    "type": "MONSTER",
+    "id": "mon_qianzhousaurus",
+    "name": { "str_sp": "Qianzhousaurus" },
+    "species": "DINOSAUR",
+    "default_faction": "qianzhousaurus",
+    "symbol": "d",
+    "color": "white_brown",
+    "volume": "5000 L",
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1263]:
```suggestion
    "volume": "550 L",
```
Match zombie and not disappear body

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 19]
```diff
@@ -1251,5 +1251,54 @@
...
     "placate_triggers": [ "MEAT" ],
     "categories": [ "DINOSAUR" ]
+  },
+  {
+    "type": "MONSTER",
+    "id": "mon_qianzhousaurus",
+    "name": { "str_sp": "Qianzhousaurus" },
+    "species": "DINOSAUR",
+    "default_faction": "qianzhousaurus",
+    "symbol": "d",
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1261]:
```suggestion
    "symbol": "D",
```
Match other dinos

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 20]
```diff
@@ -1251,5 +1251,54 @@
...
+    "color": "white_brown",
+    "volume": "5000 L",
+    "weight": "767 kg",
+    "bodytype": "elephant",
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1265]:
```suggestion
    "bodytype": "ostrich",
```
Not sure it even matters

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 21]
```diff
@@ -1251,5 +1251,54 @@
...
+    "material": [ "flesh" ],
+    "aggression": 300,
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1267]:
```suggestion
    "aggression": 90,
```
100 is max, which is zombies. This gives them some basic survival instinct, at least in theory

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 22]
```diff
@@ -1251,5 +1251,54 @@
...
+    "morale": 80,
+    "speed": 110,
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1269]:
```suggestion
    "speed": 70,
```
Still faster than Albertosaurus but slower than player

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 23]
```diff
@@ -1251,5 +1251,54 @@
...
+    "dodge": 6,
+    "melee_skill": 10,
+    "melee_dice": 10,
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1272]:
```suggestion
    "melee_dice": 4,
```
More in line with other dinos

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 24]
```diff
@@ -1251,5 +1251,54 @@
...
+    "melee_dice_sides": 8,
+    "melee_cut": 16,
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1274]:
```suggestion
    "melee_cut": 12,
```
More in line with other dinos

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 25]
```diff
@@ -1251,5 +1251,54 @@
...
+    "armor_bash": 12,
+    "armor_cut": 12,
+    "armor_bullet": 12,
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1277]:
```suggestion
    "armor_bash": 4,
    "armor_cut": 2,
    "armor_bullet": 2,
```
Match Albertosaurus

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 26]
```diff
@@ -1251,5 +1251,54 @@
...
+    "hp": 45,
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1278]:
```suggestion
    "hp": 100,
```
Similar to Ceratosaurus, closest body type I see quickly

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 27]
```diff
@@ -1251,5 +1251,54 @@
...
+    "vision_day": 36,
+    "death_function": [ "NORMAL" ],
+    "zombify_into": "mon_zianzhousaurus",
+    "description": "Also known as pinnochio rex, this bipedal dinosaur is slower than most and covered in fur, it also looks very hungry.",
+    "reproduction": { "baby_egg": "egg_qianzhousaurus", "baby_count": 5, "baby_timer": 23 },
+    "baby_flags": [ "SPRING", "SUMMER" ],
+    "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 3 },
+    "special_attacks": [ [ "Poke", 20 ], { "type": "hit", "cooldown": 10 } ],
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1286]:
```suggestion
    "special_attacks": [ { "type": "bite", "cooldown": 5 }, [ "LUNGE", 10 ] ],
```
Match Albertosaurus without its special grab. Don't think poke exists, would need to be defined.

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 28]
```diff
@@ -1251,5 +1251,54 @@
...
+    "flags": [
+      "SEES",
+      "HEARS",
+      "ANIMAL",
+      "PATH_AVOID_DANGER_1",
+      "DESTROYS",
+      "WARM",
+      "CATFOOD",
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/dinosaur.json:1294]:
```suggestion
```
Convention so far is to not make dangerous adult predators able to be tamed.

File: data/mods/DinoMod/monsters/zed-dinosaur.json [hunk 29]
```diff
@@ -835,5 +835,25 @@
...
     "upgrades": { "half_life": 14, "into": "mon_zalamosaurus" }
+  },
+  {
+    "type": "MONSTER",
+    "id": "mon_zianzhousaurus",
+    "name": { "str": "Qianz zombie" },
+    "looks_like": "mon_zyrannosaurus",
+    "volume": "550 L",
+    "weight": "770 kg",
+    "speed": 175,
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/zed-dinosaur.json:846]:
```suggestion
    "speed": 60,
```
Zombies are a little slower, about 80% speed

File: data/mods/DinoMod/monsters/zed-dinosaur.json [hunk 30]
```diff
@@ -835,5 +835,25 @@
...
+    "melee_skill": 10,
+    "melee_dice": 6,
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/zed-dinosaur.json:848]:
```suggestion
    "melee_dice": 4,
```
Match living version

File: data/mods/DinoMod/monsters/zed-dinosaur.json [hunk 31]
```diff
@@ -835,5 +835,25 @@
...
+    "melee_dice_sides": 7,
+    "melee_cut": 6,
+    "armor_bash": 10,
+    "armor_cut": 10,
+    "armor_bullet": 20,
+    "hp": 230,
+    "description": "An undead Qianzhousaurus, looking at you with a furious rage in its eyes and seeming intent on eating you.",
+    "upgrades": { "half_life": 14, "into_group": "GROUP_Zianzhousaurus_UPGRADE" },
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/zed-dinosaur.json:856]:
```suggestion
```
Either cut this or define this upgrade group and create at least two upgrades for it.

File: data/mods/DinoMod/monsters/zed-dinosaur.json [hunk 32]
```diff
@@ -835,5 +835,25 @@
...
+    "volume": "550 L",
+    "weight": "770 kg",
+    "speed": 175,
+    "melee_skill": 10,
+    "melee_dice": 6,
+    "melee_dice_sides": 7,
+    "melee_cut": 6,
+    "armor_bash": 10,
+    "armor_cut": 10,
+    "armor_bullet": 20,
```
Comment (LyleSY) [inline data/mods/DinoMod/monsters/zed-dinosaur.json:853]:
```suggestion
    "armor_bash": 5,
    "armor_cut": 5,
    "armor_bullet": 3,
```
Match zombie Albertosaurus

**Review by LyleSY — CHANGES_REQUESTED** [review]
Hey, I finally figured out how to do a review properly! Hope these make sense. Trying to work with limits of game and fit with existing dinos. Overall good stuff. If you approve these, please drop these in your local mods folder and try and spawn the regular and zombie version. Ideally they'll spawn without errors and look and act right. If they do, this is good to go.

**Comment by Theawesomeboophis** [comment]
Just added suggestions, gonna relint to properly integrate and then test.

**Comment by Theawesomeboophis** [comment]
Dinosaur.json had to be relinted. Going to test now.

**Comment by Theawesomeboophis** [comment]
Gonna retest.

**Comment by Theawesomeboophis** [comment]
DEBUG    : Error: Json error: data/mods//DinoMod/monsters/hatchling.json:685:1: requested non-existing member "symbol" in {
    "id": "mon_qianzhousaurus_hatchling",
    "type": "MONSTER",
    "name": "baby Qianzhousaurus",
    "default_faction": "qianzhousaurus",
    "upgrades": { "age_grow": 40, "into": "mon_qianzhousaurus_juvenile" }
  }


    "upgrades": { "age_grow": 40, "into": "mon_qianzhousaurus_juvenile" }
  }
]
^



 FUNCTION : bool main_menu::new_character_tab()
 FILE     : src/main_menu.cpp
 LINE     : 948
This Is what I got while testing.

**Comment by AMurkin** [comment]
Did you forget the `"copy-from"` line?

**Comment by Theawesomeboophis** [comment]
I don't know, gonna check.

**Comment by Theawesomeboophis** [comment]
Added copy_from to hatchling, gonna retest.

**Comment by Theawesomeboophis** [comment]
Loads properly now.

**Comment by Theawesomeboophis** [comment]
prehistoric creatures mod has begun production btw.

**Comment by Theawesomeboophis** [comment]
Why is continuous integration failing?

**Comment by Theawesomeboophis** [comment]
It says command exited with code 1 on appveyor.

**Comment by anothersimulacrum** [comment]
Not your fault, https://ci.appveyor.com/project/kevingranade/cataclysm-dda/builds/36255225#L704
Same thing on travis too.

**Comment by Theawesomeboophis** [comment]
> Not your fault, https://ci.appveyor.com/project/kevingranade/cataclysm-dda/builds/36255225#L704
> Same thing on travis too.
Thanks, thought I broke something again.

**Comment by Theawesomeboophis** [comment]
Gonna work on adding the other 2, then I'll work on Orcs and Ogres.

**Pull request merged**